text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/MiniMessage.java

```diff
@@ -127,6 +128,18 @@ default Component parse(final @NotNull String input) {
    */
   @NotNull Component deserialize(final @NotNull String input, final @NotNull PlaceholderResolver placeholderResolver);
 
+  /**
+   * Deserializes a string into a tree of parsed elements, with a placeholder resolver to parse placeholders of the form {@code <key>}.
+   *
+   * <p>Placeholders will be resolved from this resolver before the resolver provided in the builder is used.</p>
+   *
+   * @param input the input string
+   * @param placeholderResolver the placeholder resolver
+   * @return the root of the resulting tree
+   * @since 4.10.0
+   */
+  @NotNull ElementNode deserializeToTree(final @NotNull String input, final @NotNull PlaceholderResolver placeholderResolver);
```

Comment:
do we want to make the `ElementNode` class as an implementation detail, and expose a new `MiniMessageNode` interface, that for now just has a `toString()` method?

That would keep things buttoned down for an initial release, and allow us to expand the API more thoughtfully at a future time. Other uses of `ElementNode`, like in `Transformation`s, could also be replaced with this `MiniMessageNode`.

Reply:
What is the node used for in Transformations?

Reply:
it's used for `Modifying` type transformations, in both cases to inspect the tree to get its length for gradient purposes.

Reply:
would you not consider Nodes as they are currently as API then? those `Modifying` transformations seem to be checking for very specific classes or is that fine because it's internal?